### ISSUE TITLE
Reference doesn't take requested values

### ISSUE THREAD

**Issue by Tasselhoff** [OAI/OpenAPI-Specification#2552]
Using v3.0.2 with 42Crunch in Visual Studio Code (the issue also exists rendering in the swagger dist web directory) I have a schemas section including the two following entries:

>            destination:
>              $ref: "abstract.yml#/components/schemas/destination"
>            sender:
>              $ref: "abstract.yml#/components/schemas/ReferenceOrganization"

The former has a wrapper and also includes the latter:

>    destination: # from https://www.hl7.org/fhir/messageheader-definitions.html#MessageHeader.destination
>      type: array
>      description: the destination payer
>      items:
>        type: object
>        required:
>          - endpoint
>          - receiver
>        properties:
>          endpoint:
>            type: string
>            format: url
>          receiver:
>            $ref: "abstract.yml#/components/schemas/ReferenceOrganization"
>
>    ReferenceOrganization: # from https://www.hl7.org/fhir/organization.html 
>      type: object
>      required:
>        - identifier
>        - type
>      properties:
>        type:
>          type: string
>          default: Organization
>        identifier:
>          type: object
>          required:
>            - system
>            - value
>          properties:
>            system:
>              type: string
>            value:
>              type: string
>              description: Provider or Payer Licence

But when I use them both as follows, the destination takes the default and example from the sender:

>                destination:
>                  allOf:
>                    - $ref: "abstract.yml#/components/schemas/destination"
>                    - type: array
>                      items:
>                        properties:
>                          endpoint:
>                            example: "http://x.x.x.x/"
>                          receiver:
>                            type: object
>                            properties:
>                              identifier:
>                                properties:
>                                  system:
>                                    default: "http://y.y.y.y/license/payer-license"
>                                  value:
>                                    example: "INS-FHIR"
>                sender:
>                  allOf:
>                    - $ref: "abstract.yml#/components/schemas/ReferenceOrganization"
>                    - type: object
>                      properties:
>                        identifier:
>                          properties:
>                            system:
>                              default: "http://y.y.y.y/license/provider-license"
>                            value:
>                              example: "PR-FHIR"

Note that I also tried using allOf within destination for the receiver object, pointing to the ReferenceOrganization and it didn't change anything. What I see in the Example Value is:

>        "destination": [
>          {
>            "endpoint": "http://x.x.x.x/",
>            "receiver": {
>              "type": "Organization",
>              "identifier": {
>                "system": "http://y.y.y.y/license/provider-license",
>                "value": "PR-FHIR"
>              }
>            }
>          }
>        ],
>        "sender": {
>          "type": "Organization",
>          "identifier": {
>            "system": "http://y.y.y.y/license/provider-license",
>            "value": "PR-FHIR"
>          }
>        },

In other words, the contents of ReferenceOrganization is ignored for the destination. Am I missing something obvious here?
Update: If I put the destination after the sender in the second to last snippet, it's the destination values that are used instead, for both sender and destination.

I could just paste the contents of ReferenceOrganization into the receiver object rather than refer to the former, but that defeats the purpose of reusable objects.

No idea how to fix the formatting here, sorry (I've replaced the code with quotes which seems to render better)

**Comment by Tasselhoff** [comment]
OK, I've created a simple example, which *does* work:

`                sendOrg: 
                  allOf:
                    - $ref: "abstract.yml#/components/schemas/Org"
                    - type: object
                      properties:
                        text:
                          example: "sendOrgName"

                receiveOrg: 
                  type: array
                  items:
                    type: object
                    properties:
                      endpoint:
                        type: string
                        default: "xxx.yyy.com"
                      orgDetails:
                        allOf:
                          - $ref: "abstract.yml#/components/schemas/Org"
                          - type: object
                            properties:
                              text:
                                example: "receiveOrgName"`

calling

`    Org: 
      type: object
      properties:
        text:
          type: string
        #etc...`

gives the correct output:

`        },
        "sendOrg": {
          "text": "sendOrgName"
        },
        "receiveOrg": [
          {
            "endpoint": "xxx.yyy.com",
            "orgDetails": {
              "text": "receiveOrgName"
            }
          }
        ]`

So it appears to be due to either the nested ref or the values being for a sub-object of the referred schema.

**Comment by MikeRalphson** [comment]
I'm struggling to see what goes wrong for you and where. It does sound like you have a tooling issue, rather than an issue with the specification itself. You will probably get best results raising your issue on repositories associated with these tools.
